### ISSUE TITLE
Difference between logged metrics: loss, best_loss ?

### ISSUE THREAD

**Issue by noe** [pytorch/fairseq#737]
fairseq logs several metrics in tensorboard. Among them there is `loss` and `best_loss` . In `loss`, there is a trace for the training loss and another one for the validation loss. `best_loss` only applies to validation.

However, the values in `best_loss` are different from the minimum values of the validation `loss`, e.g. the lowest values in `loss` and much greater than the value in `best_loss` .

How are `loss` and `best_loss` computed?

**Comment by lematt1991** [comment]
Are these values also inconsistent in the textual log that gets printed to stdout?

**Comment by villmow** [comment]
> Are these values also inconsistent in the textual log that gets printed to stdout?

Yes, for me they are also different.

**Comment by myleott** [comment]
We log running averages for the epoch to the terminal [1], but point estimates to Tensorboard since it applies its own smoothing [2]. This will be more consistent with an upcoming refactoring of the logging system in fairseq.

[1] https://github.com/pytorch/fairseq/blob/master/fairseq/progress_bar.py#L63
[2] https://github.com/pytorch/fairseq/blob/master/fairseq/progress_bar.py#L278

**Comment by ischlag** [comment]
I don't think @noe's question was answered. I have the same question.
> How are loss and best_loss computed?

@myleott Could you explain the difference between the best_loss and loss metric in tensorboard? And where I can find that in the code? I tried looking for it but didn't find it.
